### PR TITLE
[explorer] validators map

### PR DESCRIPTION
**Pull request by Nikhil-Mysten**
## Description 

image

## Test Plan 

Locally

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [x] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes
New Validators Map on Home Page

### CODE REVIEW

File: apps/explorer/src/components/validator-map/index.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+}
+
+// NOTE: This component is lazy imported, so it needs to be default exported:
+export default function ValidatorMap({ minHeight }: Props) {
+    // TODO BEFORE MERGING: update query URI
+    const { data, isLoading, isSuccess } = useQuery(
+        ['validator-map'],
+        async () => {
+            const res = await fetch(
+                `http://localhost:3003/validator-map?network=testnet`,
```
Comment (Nikhil-Mysten) [inline apps/explorer/src/components/validator-map/index.tsx:48]: ignore - just waiting to verify the deployed apps-be piece to finalize this

File: apps/explorer/src/components/validator-map/WorldMap.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import { type ValidatorMapData } from './types';
+
+// @ts-expect-error: The types of `world` here aren't aligned but they are correct
+const land = topojson.feature(world, world.objects.countries) as unknown as {
+    type: 'FeatureCollection';
+    features: Feature[];
+};
+
+// We hide Antarctica because there will not be validators there:
+const HIDDEN_REGIONS = ['Antarctica'];
```
Comment (mamos-mysten) [inline apps/explorer/src/components/validator-map/WorldMap.tsx:21]: this is interesting!

File: apps/explorer/src/components/validator-map/index.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+import { useQuery } from '@tanstack/react-query';
+import { ParentSize } from '@visx/responsive';
+import { TooltipWithBounds, useTooltip } from '@visx/tooltip';
+import React, { type ReactNode, useCallback, useMemo } from 'react';
+
+import { WorldMap } from './WorldMap';
+import { type ValidatorMapData } from './types';
+
+// import { useNetwork } from '~/context';
```
Comment (mamos-mysten) [inline apps/explorer/src/components/validator-map/index.tsx:12]: can remove

File: apps/explorer/src/pages/home/Home.tsx [hunk 4]
```diff
@@ -19,7 +19,7 @@ import { SuiTokenCard } from '~/components/SuiTokenCard';
 import { Card } from '~/ui/Card';
 import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '~/ui/Tabs';
 
-const NodeMap = lazy(() => import('../../components/node-map'));
```
Comment (Jordan-Mysten) [inline apps/explorer/src/pages/home/Home.tsx:22]: can you please remove this while you're at this? Don't want duplicated code and it's harder to review diffs that are all additive

**Review by Jordan-Mysten — CHANGES_REQUESTED** [review]

File: apps/explorer/src/components/validator-map/WorldMap.tsx (hunk 2 above)
Comment (Nikhil-Mysten) [reply apps/explorer/src/components/validator-map/WorldMap.tsx:21]: a lot of this is copy/paste 😅

File: apps/explorer/src/components/validator-map/index.tsx (hunk 1 above)
Comment (Jordan-Mysten) [reply apps/explorer/src/components/validator-map/index.tsx:48]: `useAppsBackend` should give you a fetch function you can use

File: apps/explorer/src/components/validator-map/WorldMap.tsx (hunk 2 above)
Comment (mamos-mysten) [reply apps/explorer/src/components/validator-map/WorldMap.tsx:21]: did we lose the D/W/M/ALL filters? are we planning on adding them back? don't have a lot of context here, just curious

Comment (Nikhil-Mysten) [reply apps/explorer/src/components/validator-map/WorldMap.tsx:21]: they don't exist on the new design https://www.figma.com/file/kSD0qXCIDZA4KQVNepA5hR/02-Explorer?type=design&node-id=7077-248095&t=SwZDDyCOuicy4adn-0

File: apps/explorer/src/pages/home/Home.tsx (hunk 4 above)
Comment (Nikhil-Mysten) [reply apps/explorer/src/pages/home/Home.tsx:22]: @mystie711 are we trying to replace the node map on the validators page as well? or should we keep that as is

File: apps/explorer/src/components/validator-map/index.tsx (hunk 1 above)
Comment (mamos-mysten) [reply apps/explorer/src/components/validator-map/index.tsx:48]: Did `useAppsBackend` not work out here or are we waiting for the PR to land there before updating?

File: apps/explorer/src/components/validator-map/types.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+export interface Feature {
+    type: 'Feature';
+    id: string;
+    geometry: { coordinates: [number, number][][]; type: 'Polygon' };
+    properties: { name: string; alpha2: string };
```
Comment (mamos-mysten) [inline apps/explorer/src/components/validator-map/types.ts:8]: What does `alpha2` represent? Country code?

**Review by mamos-mysten — APPROVED** [review]
Two small comments but LGTM

File: apps/explorer/src/components/validator-map/types.ts (hunk 5 above)
Comment (Nikhil-Mysten) [reply apps/explorer/src/components/validator-map/types.ts:8]: yes, seems to be country code

Comment (Nikhil-Mysten) [reply apps/explorer/src/components/validator-map/types.ts]: renamed it, doesn't seem to be used anywhere else

**Comment by mystie711** [comment]
@Nikhil-Mysten could we update the padding on top and bottom left of the text to the figma values? This way all cards will look balanced on the home page.

**Pull request merged**